### ISSUE TITLE
False negative in SC2155

### ISSUE THREAD

**Issue by lvjp** [koalaman/shellcheck#1733]
#### For bugs
- Rule Id: SC2155
- My shellcheck version: 0.7.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2155)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
set -o errexit
export FOO="$(myEnvBuilder)"
FOO="$(myEnvBuilder)" ./myCmd.sh
```
#### Here's what shellcheck currently says:

```
In myscript.sh line 3:
export FOO="$(myEnvBuilder)"
       ^-^ SC2155: Declare and assign separately to avoid masking return values.
```

#### Here's what I wanted or expected to see:

```
In myscript.sh line 3:
export FOO="$(myEnvBuilder)"
       ^-^ SC2155: Declare and assign separately to avoid masking return values.

In myscript.sh line 4:
FOO="$(myEnvBuilder)" ./myCmd.sh
^-^ SC2155: Declare and assign separately to avoid masking return values.
```

In the line 4, the return value of `myEnvBuilder` is effectively ignored but not detect as a bug within shellcheck.

**Comment by matthewpersico** [comment]
Can you simplify the sample? I am betting that the ‘export’ has nothing to do with it; it’s probably the addition of the string that confuses shellcheck. What happens if you eliminate the export line totally (I’d try it myself but I’m on my iPhone in an IHOP).

**Comment by lvjp** [comment]
@matthewpersico I edit the issue to have a simpler example.

**Comment by lvjp** [comment]
Thank you @jabberabbe, you commit jabberabbe/shellcheck@be0d0247b180f0da68edac57ae1498b90eb2ff30 did the job.

Can't wait for pull request #1739 to be approuved.

**Comment by koalaman** [comment]
What's your suggested rewrite for this?
